scripts/2_explore_relationship_AandB_pt2.ipynb

```diff
@@ -0,0 +1,3152 @@
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Define function to apply latent space transformation and output reconstructed data\n",
+    "\n",
+    "def interpolate_in_vae_latent_space_AB(all_data, \n",
+    "                                       sample_data,\n",
+    "                                       model_encoder_file,\n",
+    "                                       model_decoder_file,\n",
+    "                                       weights_encoder_file,\n",
+    "                                       weights_decoder_file,\n",
+    "                                       encoded_dir,\n",
+    "                                       gene_id,\n",
+    "                                       percent_low,\n",
+    "                                       percent_high,\n",
+    "                                       out_dir):\n",
+    "    \"\"\"\n",
+    "    interpolate_in_vae_latent_space(all_data: dataframe,\n",
+    "                                    sample_data: dataframe,\n",
+    "                                    model_encoder_file: string,\n",
+    "                                    model_decoder_file: string,\n",
+    "                                    weights_encoder_file: string,\n",
+    "                                    weights_decoder_file: string,\n",
+    "                                    encoded_dir: string,\n",
+    "                                    gene_id: string,\n",
+    "                                    percent_low: integer,\n",
+    "                                    percent_high: integer,\n",
+    "                                    out_dir: string):\n",
+    "\n",
+    "    input:\n",
+    "        all_data: Dataframe with gene expression data from all samples\n",
+    "        \n",
+    "        sample_data:  Dataframe with gene expression data from subset of samples (around the treshold)\n",
+    "\n",
+    "        model_encoder_file: file containing the learned vae encoder model\n",
+    "\n",
+    "        model_decoder_file: file containing the learned vae decoder model\n",
+    "        \n",
+    "        weights_encoder_file: file containing the learned weights associated with the vae encoder model\n",
+    "        \n",
+    "        weights_decoder_file: file containing the learned weights associated with the vae decoder model\n",
+    "        \n",
+    "        encoded_dir:  directory to use to output offset vector to \n",
+    "\n",
+    "        gene_id: gene you are using as the \"phenotype\" to sort samples by \n",
+    "\n",
+    "                 This gene is referred to as \"target_gene\" in comments below\n",
+    "\n",
+    "\n",
+    "        percent_low: integer between 0 and 1\n",
+    "\n",
+    "        percent_high: integer between 0 and 1\n",
+    "        \n",
+    "        out_dir: directory to output predicted gene expression to\n",
+    "\n",
+    "    computation:\n",
+    "        1.  Sort samples based on the expre
```

Comment:
Is this file used in this function here? Does this function perform interpolation calculations? If so I don't see the code that does the transforming in this function.

Reply:
yes, this interpolation function is performing the interpolation
```or sample_id in sample_ids:
        intermediate_target_gene_exp = target_gene_sorted[sample_id]
        print('gene A exp is {}'.format(intermediate_target_gene_exp))
        alpha = get_scale_factor(
            target_gene_sorted, intermediate_target_gene_exp, percent_low, percent_high)
        print('scale factor is {}'.format(alpha))
        predict = baseline_encoded + alpha * offset_encoded```

For each sample it looks at where that sample lies along the phenotype gradient and that gives you your alpha.  Then it predicts a new gene expression for that sample by saying 
new exp = baseline exp + alpha*offset vector

where this baseline exp is the average lowest level of expression for the given phenotype

Hopefully that makes sense.  If not I can explain better in person